### PR TITLE
Add a few useful os calls to nimscript

### PR DESCRIPTION
**Pull request by genotrance**
Added the following procs to nimscript:-
- cpDir() = copyDir() - #7393
- mvDir() = moveDir()
- pwd() = getCurrentDir()
- where() = findExe()

I was going to add some test cases into tests/newconfig/tfoo.nims but realized the `nims` extension is not covered by `testament`.

### CODE REVIEW

**Comment by data-man** [comment]
Maybe to rename a functions?
cpDir() -> copyDir()
mvDir() -> moveDir()
pwd() -> getCurrentDir()
where() -> findExe()

For compatibility with stdlib.
And ```pwd()``` for Windows-users it's not clear. And ```where()``` for *x-users. :)

**Comment by genotrance** [comment]
@Araq preferred different names which is why I did it this way.

https://irclogs.nim-lang.org/29-03-2018.html#23:18:36

File: lib/system/nimscript.nim [hunk 1]
```diff
@@ -206,12 +210,24 @@ proc mvFile*(`from`, to: string) {.raises: [OSError].} =
     moveFile `from`, to
     checkOsError()
 
+proc mvDir*(`from`, to: string) {.raises: [OSError].} =
+  ## Moves the dir `from` to `to`.
+  log "mvDir: " & `from` & ", " & to:
+    moveDir `from`, to
+    checkOsError()
```
Comment (dom96) [inline lib/system/nimscript.nim]: This is unrelated to your PR, but why is this ``checkOsError`` needed? Won't movedDir etc. raise OSError?

**Comment by dom96** [comment]
> @Araq preferred different names which is why I did it this way.

It has just occurred to me why perhaps following Rust's convention might make sense, exclamation mark at the end of every compile-time proc.

File: lib/system/nimscript.nim (hunk 1 above)
Comment (genotrance) [reply lib/system/nimscript.nim]: The exceptions are caught in `template cbos()` and errorMsg is set there for `checkOsError()` to check and re-raise it.

**Comment by Araq** [comment]
1. Rust uses the exclamation mark for macros.
2. Exclamation marks are ugly and patronizing. Nim is not patronizing.

**Comment by Araq** [comment]
Sorry this took so long to review. The names are wrong. NimScript names differ from the stdlib names iff they do different things. ``rmDir`` is not an alias for ``removeDir`` it logs its action and supports ``ScriptMode.Whatif``. But your ``where`` and ``pwd`` are just aliases, these should be named ``findExe`` and ``getCurrentDir``. Btw ``getCurrentDir`` is already supported in ``nimscript.nim``.

**Comment by genotrance** [comment]
I noticed `getCurrentDir()` but it isn't exported. I also think it is fair to have unique names in nimscript so that once `import os` does work in nimscript, it won't be a proc name conflict. Especially for backward compatibility, we won't be able to remove getCurrentDir() from nimscript as existing scripts will break without "import os".

I can change where() and pwd() back to their original names if there's a way to avoid the conflict in the future without requiring any script changes.

**Comment by Araq** [comment]
Well currently you cannot import os in NimScript and once you can you need to disambiguate via ``import os except getCurrentDir`` or similar. Not ideal but better than changing the reasoning behind these names.

**Comment by genotrance** [comment]
Added to changelog under "Library additions".

**Pull request merged**